### PR TITLE
[BugFix]  ignore checking its close state before Prepare

### PR DESCRIPTION
**Pull request by fzhedu**
## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5

### CODE REVIEW

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 1]
```diff
@@ -700,8 +701,13 @@ Status FragmentExecutor::execute(ExecEnv* exec_env) {
     {
         SCOPED_TIMER(prepare_instance_timer);
         SCOPED_TIMER(prepare_driver_timer);
-        RETURN_IF_ERROR(_fragment_ctx->iterate_drivers(
-                [state = _fragment_ctx->runtime_state()](const DriverPtr& driver) { return driver->prepare(state); }));
+        Status st;
+        TRY_CATCH_ALL(st,
```
Comment (stdpain) [inline be/src/exec/pipeline/fragment_executor.cpp:705]: why we need catch exception here?

Comment (fzhedu) [reply be/src/exec/pipeline/fragment_executor.cpp:706]: what if here has exceptions? it will kill the process.

**Comment by rickif** [comment]
@fzhedu Please backport this PR to branch-3.1 and branch-3.2.

File: be/src/exec/pipeline/fragment_executor.cpp (hunk 1 above)
Comment (satanson) [reply be/src/exec/pipeline/fragment_executor.cpp:706]: prepare method just return error status when it fails.
no need to catch exception.

**Review by satanson — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Pull request merged**